File: Easy_Problems/3. Sorted Square Array/Javascript/Solution1.js
// O(nlog(n)) time | O(n) space
//  Brute Force Approach

function sortSquareArray(array) {
    arr_l = array.length
    sorted_arr = Array(arr_l)

    for (let i = 0; i < arr_l; i++) {
        sorted_arr[i] = Math.pow(array[i], 2)
    }

    sorted_arr.sort((a, b) => a - b)
    
    return sorted_arr;
}

sortSquareArrayTest = () => {
    console.log(sortSquareArray([-4, -1, 0, 3, 10])) // [0, 1, 9, 16, 100]
    console.log(sortSquareArray([-7, -3, 2, 3, 11])) // [4, 9, 9, 49, 121]
    console.log(sortSquareArray([-2, -1, 0, 2, 10])) // [0, 1, 4, 4, 100]
    console.log(sortSquareArray([0, 1, 4, 6, 7, 9])) // [0, 1, 16, 36, 49, 81]
}

File: Easy_Problems/3. Sorted Square Array/Java/Solution1.java
import java.util.Arrays;

class Solution1{

    // O(nlog(n)) time | O(n) space
    // Brute Force Approach
    private static int[] sortSquareArray(int[] array) {
        int arr_l = array.length;
        int[] sorted_arr = new int[arr_l];
        
        for (int i = 0; i < arr_l; i++) {
            sorted_arr[i] = array[i] * array[i];
        }
        Arrays.sort(sorted_arr);
        return sorted_arr;
    }
    
}